contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checker.py

```diff
@@ -167,6 +169,17 @@ def checkstyle(self, sources):
 
   def execute(self):
     """Run Checkstyle on all found non-synthetic source files."""
+
+    # If we are linting for Python 3, skip lint altogether. 
+    # Long-term Python 3 linting solution tracked by:
+    # https://github.com/pantsbuild/pants/issues/5764
+    intepreter = self.context.products.get_data(PythonInterpreter)
+    if intepreter and intepreter.version > (3, 0 ,0):
```

Comment:
Products should never be None if your task is executing (the upstream task would have failed and your task's execution should short circuit). You should drop the `None` test and add this: https://github.com/pantsbuild/pants/blob/58a38dd135366f3533ce315004340fdcd0d03a59/src/python/pants/backend/python/tasks/resolve_requirements.py#L22-L24.